### ISSUE TITLE
False positive when a null coalesce operator is used

### ISSUE THREAD

**Issue by morozov** [vimeo/psalm#210]
Inspecting the following code with the default configuration:
```php
function fn1(?string $var) : ?string
{
    return $var;
}

function fn2() : string
{
    return 'bar';
}

function fn3(string $var) : void
{
}

$x = fn1('foo') ?? fn2();
fn3($x);
```
on 0.3.55 produces:
```
↪  psalm 
ERROR: PossiblyNullArgument - src/issue-1.php:18:5 - Argument 1 of fn3 cannot be null, possibly null value provided
fn3($x);

Checks took 0.04902720451355 and used 14.234MB
```
It doesn't look correct since in the case if `fn1()` returns `NULL`, `$x` will be assigned the return value of `fn2()` which cannot be `NULL`. Thus, `$x` cannot be `NULL` in any case.

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/5b3531b5a50c63e54dbde1e27be90a43fe24c25f

Thank for the good testcase!

**Comment by morozov** [comment]
Wow, that's… fast. Thanks!
